api/events/schema.py

```diff
@@ -35,6 +39,80 @@ class Meta:
         model = EventsModel
 
 
+class BookEvent(graphene.Mutation):
+    """
+        Book Calendar events
+    """
+    response = graphene.String()
+
+    class Arguments:
+        event_title = graphene.String(required=True)
+        start_date = graphene.String(required=True)
+        start_time = graphene.String(required=True)
+        duration = graphene.Float(required=False)
+        attendees = graphene.String(required=False)
+        organizer = graphene.String(required=False)
+        description = graphene.String(required=False)
+        room = graphene.String(required=False)
+        time_zone = graphene.String(required=False)
+
+    @Auth.user_roles('Admin', 'Default User', 'Super Admin')
+    def mutate(self, info, **kwargs):
+        """Creates calendar events
+
+        Args:
+            event_title: A sting that communicates the event summary/title [required]
+            start_date: The start date of the event eg. 'Nov 4 2019' [required]
+            start_time: The end time of the event eg '07:00 AM' [required]
+            duration: A float of the duration of the event
```

Comment:
I can not see the event end time, does it mean that the end time will be generated using this 'duration field'?

Reply:
Yes @mifeille, I created a helper that computes the event end time by adding the event duration to the start time provided by the user.